packages/dds/tree/src/test/tablePerformanceTestUtilities.ts

```diff
@@ -0,0 +1,190 @@
+
+/**
+ * Define a return type for table tree creation.
+ */
+export interface TableTreeDefinition {
+	/**
+	 * The table tree instance.
+	 */
+	table: InstanceType<typeof Table>;
+	/**
+	 * The undo stack for the table tree.
+	 */
+	undoStack: Revertible[];
+	/**
+	 * The redo stack for the table tree.
+	 */
+	redoStack: Revertible[];
+	/**
+	 * Unsubscribe from the table tree events and dispose of the undo/redo stacks.
+	 */
+	unsubscribe: () => void;
+}
+
+/**
+ * Factory for creating a table tree schema.
+ * This factory is used to create the schema for the table tree, including cells, columns,
+ * rows, and the table itself.
+ */
+const schemaFactory = new SchemaFactoryAlpha("test");
+
+/**
+ * Defines the schema for a table cell.
+ */
+export class Cell extends schemaFactory.object("table-cell", {
+	cellValue: schemaFactory.string,
+}) {}
+
+/**
+ * Defines the schema for a table column.
+ */
+export class Column extends TableSchema.column({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+/**
+ * Defines the schema for a table row.
+ */
+export class Row extends TableSchema.row({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+/**
+ * Defines the schema for a table, which includes columns and rows.
+ * It uses the previously defined Cell, Column, and Row schemas.
+ */
+export class Table extends TableSchema.table({
+	schemaFactory,
+	cell: Cell,
+	column: Column,
+	row: Row,
+}) {}
+
+/**
+ * Provides a simple table tree initialized with the specified size and cell value.
+ * This helper function creates a table schema, initializes a SharedTree instance,
+ * and populates it with the specified number of rows and columns.
+ * Each cell is initialized with the provided cell value.
+ *
+ * @param tableSize - The number of rows and columns to create in the table.
+ * @param cellValue - The initial value to set in each cell of the table.
+ * @returns A fully initialized table tree definition, including table instance, undo/redo stacks, and a cleanup function.
+ */
+export function createTableTree(tableSize: number, cellValue: string): TableTreeDefinition {
+	const sharedTreeFactory = DefaultTestSharedTreeKind.getFactory();
+	const runtime = new MockFluidDataStoreRuntime({
+		idCompressor: createIdCompressor(),
+		attachState: AttachState.Detached,
+	});
+	const tree = sharedTreeFactory.create(runtime, "tree");
+	const treeView = tree.viewWith(
+		new TreeViewConfiguration({
+			schema: Table,
+			enableSchemaValidation: true,
+		}),
+	);
+
+	treeView.initialize(Table.empty());
+	const undoStack: Revertible[] = [];
+	const redoStack: Revertible[] = [];
+
+	function onDispose(disposed: Revertible): void {
+		const redoIndex = redoStack.indexOf(disposed);
+		if (redoIndex !== -1) {
+			redoStack.splice(redoIndex, 1);
+		} else {
+			const undoIndex = undoStack.indexOf(disposed);
+			if (undoIndex !== -1) {
+				undoStack.splice(undoIndex, 1);
+			}
+		}
+	}
+
+	const unsubscribeFromCommitAppliedEvent = treeView.events.on(
```

Comment:
Subscribing _before_ filling the tree will result in us building up a large revertible stack from the column/row/cell insertions that were part of initialization. I don't think that's what we want. The undo/redo tests have specific operations they want inserted into the stack(s).

We also probably don't want to incur the overhead of these subscriptions during the measurements. I think my recommendation would be:
1. Don't track undo/redo stacks outside of undo/redo tests.
2. Only start populating the undo/redo stacks immediately before we do the pre-test operations that we plan to undo/redo
3. Unsubscribe from event after we've built up the stack(s) we need for the test / before we begin measurement.

Reply:
Updated. The undo/redo stack tracking now starts immediately before the pre-test operations to ensure only relevant operations are added to the stacks. Subscriptions to the `commitApplied` event are also unsubscribed after the stacks are populated to avoid unnecessary overhead during measurement. Additionally, the redo stack is cleared on non-undo commits to prevent stale operations from being reused.